src/classes/RD2_ApiService.cls

```diff
@@ -253,4 +258,248 @@ public inherited sharing class RD2_ApiService {
             currencyCode = currencyCode = RD2_ApiService.getCurrencyCode(rd);
         }
     }
+
+    public String pause(Map<Id, PauseObject> pauseData) {
+        if (!RD2_EnablementService.isRecurringDonations2Enabled) {
+            return System.Label.RD2_EnhancedRecurringDonationsNotEnabled;
+        }
+
+        if (pauseData.size() > MAXIMUM_RECORDS_TO_PROCESS) {
+            return String.format(System.Label.CallableApiMaximumRecordsExceeded,
+                new List<String> { String.valueOf(MAXIMUM_RECORDS_TO_PROCESS) });
+        }
+
+        return JSON.serialize(handlePauseRequests(pauseData));
+    }
+
+    private Map<Id, PauseObject> handlePauseRequests(Map<Id, PauseObject> pauseData) {
+        List<RecurringDonationSchedule__c> schedulesToInsert = new List<RecurringDonationSchedule__c>();
+        List<RecurringDonationSchedule__c> schedulesToUpdate = new List<RecurringDonationSchedule__c>();
+
+        Map<Id, npe03__Recurring_Donation__c> recurringDonationsById = new Map<Id, npe03__Recurring_Donation__c>(
+            getRecurringDonationsById(pauseData.keySet())
+        );
+
+        for (Id rdId : pauseData.keySet()) {
+            RD2_RecurringDonation rdRecord = new RD2_RecurringDonation(recurringDonationsById.get(rdId));
+            PauseObject pauseInfo = pauseData.get(rdId);
+
+            populateDefaultValue(pauseInfo);
```

Comment:
```suggestion
            PauseObject pauseInfo = pauseData.get(rdId);

            pauseInfo.populateDefaults();
```

See this comment: https://github.com/SalesforceFoundation/NPSP/pull/6469/files#r648729887